cloud-init: Support injection of additional SSH keys on instances (#14959)
canonical/lxd: doc/api-extensions.md

1. {config:option}`storage-pure-pool-conf:pure.api.token`
1. {config:option}`storage-pure-pool-conf:pure.mode`
1. {config:option}`storage-pure-pool-conf:pure.target`

## `cloud_init_ssh_keys`

> Adds support for injecting additional SSH public keys into instances through `cloud-init` without conflicting with any configuration present on `cloud-init.vendor-data` or `cloud-init.user-data`.

Reviewer: ```suggestion
Adds support for injecting additional SSH public keys into instances through {ref}`cloud-init <instance-options-cloud-init>` without conflicting with any configuration present on {config:option}`instance-cloud-init:cloud-init.vendor-data` or {config:option}`instance-cloud-init:cloud-init.user-data`.
```

Author: tbh I wasn't aware of this syntax used in `{config:option}`instance-cloud-init:cloud-init.user-data` until now. Thanks for catching this and I will pay attention to it from now on.

1. {config:option}`storage-pure-pool-conf:pure.api.token`
1. {config:option}`storage-pure-pool-conf:pure.mode`
1. {config:option}`storage-pure-pool-conf:pure.target`

## `cloud_init_ssh_keys`

Adds support for injecting additional SSH public keys on instances through `cloud-init` without conflicting with any configuration present on `cloud-init.vendor-data` or `cloud-init.user-data`.

> To achieve this, the `cloud-init.ssh-keys.KEYNAME` configuration key is added for both instances and profiles, that can be used for specifying one desired additional public key to be injected. Where `KEYNAME` can be any arbitrary name for the injected key.

Reviewer 2: Please can you document here what the value of this key should be too (as well as in the docs).

> To achieve this, the `cloud-init.ssh-keys.KEYNAME` configuration key is added for both instances and profiles, that can be used for specifying one desired additional public key to be injected. Where `KEYNAME` can be any arbitrary name for the injected key.

Reviewer: ```suggestion
To achieve this, the `cloud-init.ssh-keys.KEYNAME` configuration key is added for both instances and profiles, used to define a public key to be injected. `KEYNAME` can be any arbitrary name for the injected key.
```
